cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaConnectBuildTest.java

```diff
@@ -226,7 +230,78 @@ public void testDeployment()   {
     }
 
     @Test
-    public void testDeploymentWithoutPushSecret()   {
+    public void testBuildahDeployment() {
+        Map<String, Quantity> limit = new HashMap<>();
+        limit.put("cpu", new Quantity("500m"));
+        limit.put("memory", new Quantity("512Mi"));
+
+        Map<String, Quantity> request = new HashMap<>();
+        request.put("cpu", new Quantity("1000m"));
+        request.put("memory", new Quantity("1Gi"));
+
+        KafkaConnect kc = new KafkaConnectBuilder()
+            .withNewMetadata()
+                .withName(cluster)
+                .withNamespace(namespace)
+            .endMetadata()
+            .withNewSpec()
+                .withImage("my-source-image:latest")
+                .withBootstrapServers("my-kafka:9092")
+                .withNewBuild()
+                    .withNewDockerOutput()
+                        .withImage("my-image:latest")
```

Comment:
Would it be better to test for the default image instead? That does not seem to be covered anywhere (or I missed it)

Reply:
What default image do you mean? IIRC there is no default image for the output field -> because it's required field so users can set the destination where the image should be pushed. Or do you mean checking if the error is thrown when here and in `.spec.image` is set the same image?

Reply:
Sorry, I guess I commented on a wrong line. But what I wonder is whether you test soemwhere that the right image is used for the Build Pod/container.

Reply:
I will check it, I think there is none actually. Good catch, thanks :)

Reply:
So there is a check for the image used in the Build Pod:
```
assertThat(pod.getSpec().getContainers().get(0).getImage(), is(build.image));
```
and it's present in both tests for Kaniko and Buildah.
However, we cannot test change of the image as it would mean changing the env variables, which is not so easy in UTs.
I think it's sufficient WDYT?

Reply:
And if nothing else, you have the default image there, or?